honeybee_3dm/config.py

```diff
@@ -233,200 +199,23 @@ def check_config(file_3dm, config_path):
-        hb_face.properties.radiance.modifier = radiance_modifiers[config
-            ['layers'][layer_name]['radiance_material']]
-        return hb_face
-    else:
-        return hb_face
-
-
-def face3d_to_hb_face_with_rad(config, face_obj, name, layer_name):
-    """Create a Honeybee Face object with a radiance material assigned to it.
-
-    Args:
-        config: A dictionary of the config settings.
-        face_obj: A Ladybug Face3d object.
-        name: A text string of the name of the rhino object.
-        layer_name: A text string of the rhino layer name.
-
-    Returns:
-        A Honeybee Face object.
-    """
-    
-    obj_name = name or clean_and_id_string(layer_name)
-    args = [clean_string(obj_name), face_obj]
-    hb_face = Face(*args)
-    hb_face.display_name = args[0]
-    
-    if 'radiance_material' in config['layers'][layer_name]:
-        radiance_modifiers = mat_to_dict(config['sources']['radiance_material'])
-        hb_face.properties.radiance.modifier = radiance_modifiers[config
-            ['layers'][layer_name]['radiance_material']]
-        return hb_face
+    try:
+        with open(config_path) as fh:
+            config = json.load(fh)
+    except json.decoder.JSONDecodeError:
+        raise ValueError(
+            'Not a valid json file.'
+            )
     else:
-        return hb_face
-
-
-def face3d_to_hb_object(config, face_obj, name, layer_name):
-    """Create Honeybee Aperture, Shade, and Door objects.
-
-    Args:
-        config: A dictionary of the config settings.
-        face_obj: A Ladybug Face3d object.
-        name: A text string of the name of the rhino object.
-        layer_name: A text string of the rhino layer name.
-
-    Returns:
-        A tuple of lists. Each list contains Honeybee Aperture objects, 
-            Honeybee Shade objects, and Honeybee Door objects. List will 
-            be empty if no objects are found for that Honeybee object.
-    """
-
-    hb_shades, hb_apertures, hb_doors = tuple([[] for _ in range(3)])
-
-    obj_name = name or clean_and_id_string(layer_name)
-    args = [clean_string(obj_name), face_obj]
-    
-    if config['layers'][layer_name]['honeybee_face_object'] == 'aperture':
-        hb_aperture = Aperture(*args)
-        hb_aperture.display_name = args[0]
+        # Validate against schema
+        config_obj = Config.parse_file(config_path)
+        config = config_obj.dict(exclude_none=True)
         
-        if 'radiance_material' in config['layers'][layer_name]:
-            radiance_modifiers = mat_to_dict(config['sources']['radiance_material'])
-            hb_aperture.properties.radiance.modifier = radiance_modifiers[config[
-                'layers'][layer_name]['radiance_material']]
-            hb_apertures.append(hb_aperture)
+        # Validate config with external binaries
+        if check_layers(file_3dm, config) and check_rad(config):
+            return config
         else:
```

Comment:
Doesn't the two checks raise and error if something is wrong? Do you need the else clause here?